Source/WebKit/WebProcess/Extensions/Cocoa/WebExtensionContextProxyCocoa.mm

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if !__has_feature(objc_arc)
+#error This file requires ARC. Add the "-fobjc-arc" compiler flag for this file.
+#endif
+
+#include "config.h"
+#include "WebExtensionContextProxy.h"
+
+#if ENABLE(WK_WEB_EXTENSIONS)
+
+#include "WebExtensionAPINamespace.h"
+#include "WebExtensionAPIWebNavigation.h"
+#include <WebCore/ProcessQualified.h>
+#include <wtf/ObjectIdentifier.h>
+
+namespace WebKit {
+
+using namespace WebCore;
+
+// MARK: webNavigation support
```

Comment:
I've seen both `Mark:` and `Mark: - `. Is there a styling preference of using one over the other?

Reply:
They have slightly different behaviors in Xcode's UI.

`MARK: -` puts a separator above the item, `MARK:` doesn't.

without -:
<img width="641" alt="Screenshot 2022-12-20 at 10 25 45 AM" src="https://user-images.githubusercontent.com/114437188/208739254-cc663d1b-0428-4510-ae76-3696dab8308b.png">

vs with -:
<img width="582" alt="Screenshot 2022-12-20 at 10 26 45 AM" src="https://user-images.githubusercontent.com/114437188/208739414-08f691da-918b-4f28-a1db-d3965b103ba8.png">

Reply:
I see. Good to know. Thanks!!